apps/portals-base/src/vite.config.ts

```diff
@@ -0,0 +1,118 @@
+import { vitestConfig } from 'vite-config'
+import { mergeConfig } from 'vitest/config'
+import { createHtmlPlugin } from 'vite-plugin-html'
+
+export default mergeConfig(vitestConfig, {
+  envDir: './src/config',
+  test: {
+    include: ['src/tests/**/*.test.[jt]s?(x)'],
+    setupFiles: ['src/tests/setupTests.ts'],
+    environment: 'jsdom', // introduced due to random "ReferenceError: window is not defined" during tests
+  },
+  plugins: [
+    createHtmlPlugin({
```

Comment:
Use `vite-plugin-html` to inject shared HTML into the portals sites, giving us one source of truth.